### PR TITLE
sparcv8leon3: fix interrupts/reschedule

### PR DESCRIPTION
**Pull request by lukileczo**
## Description

This change:
- fixes wrong psr value being set after interrupt/reschedule
- creates separate cpu context for global/out and local/in registers
- unifies register names in asm code
## Motivation and Context

DONE: JIRA: RTOS-516
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `sparcv8leon3-gr716-mini`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/sparcv8leon3/arch/cpu.h [hunk 1]
```diff
@@ -81,6 +81,29 @@
...
+	/* in */
+	u32 i0;
+	u32 i1;
+	u32 i2;
+	u32 i3;
+	u32 i4;
+	u32 i5;
+	u32 fp;
+	u32 i7;
+} cpu_win_context_t;
```
Comment (agkaminski) [inline hal/sparcv8leon3/arch/cpu.h:104]: As `win` is not a separate module within the kernel, camelCase should be used

```suggestion
} cpu_winContext_t;
```

File: hal/sparcv8leon3/cpu.c [hunk 2]
```diff
@@ -58,22 +74,22 @@ int hal_cpuCreateContext(cpu_context_t **nctx, void *start, void *kstack, size_t
...
+	wctx->l0 = 0xeeeeeee0;
+	wctx->l1 = 0xeeeeeee1;
+	wctx->l2 = 0xeeeeeee2;
+	wctx->l3 = 0xeeeeeee3;
+	wctx->l4 = 0xeeeeeee4;
+	wctx->l5 = 0xeeeeeee5;
+	wctx->l6 = 0xeeeeeee6;
+	wctx->l7 = 0xeeeeeee7;
+
+	wctx->i0 = 0x10000000;
```
Comment (agkaminski) [inline hal/sparcv8leon3/cpu.c:86]: How is `arg` now passed to the thread?

Comment (lukileczo) [reply hal/sparcv8leon3/cpu.c:86]: in `o0` register (`ctx->o0`)

Comment (agkaminski) [reply hal/sparcv8leon3/cpu.c:86]: Oh, ok, I missed the fact it was assigned two times

**Review by agkaminski — APPROVED** [review]
LGTM

**Pull request merged**